internal/primitives/database/mem.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package database
+
+import (
+	"sync"
+
+	"github.com/ChainSafe/gossamer/internal/primitives/core/hash"
+	"github.com/ChainSafe/gossamer/internal/primitives/runtime"
+)
+
+type countValue struct {
+	Count uint32
+	Value []byte
+}
+
+// MemDB implements `Database` as an in-memory hash map. `Commit` is not atomic.
+type MemDB[H runtime.Hash] struct {
+	inner map[ColumnID]map[string]countValue
+	sync.RWMutex
+}
+
+// NewMemDB is constructor for MemDB
+func NewMemDB[H runtime.Hash]() *MemDB[H] {
+	return &MemDB[H]{
+		inner: make(map[ColumnID]map[string]countValue),
```

Comment:
should mutex be created here as well?

Reply:
how do you create a mutex? (there is no constructor btw)